### PR TITLE
Fix comment parsing with unattached comments

### LINKED ISSUE #80: Error on __schema query with comments
I am using the latest `3.0.0-preview-1631` so parser version is  `5.1.2` . The following query gives an error :

```javascript
{
  __schema {
    
    types {
      kind
      
    }
  }
  
# __schema{
#   types {
#     name
#   }
# }
}
```

The error is :  
```javascript
{
  "errors": [
    {
      "message": "GraphQL.ExecutionError: ParserContext has 1 not applied comments.\r\n ---> System.ApplicationException: ParserContext has 1 not applied comments.\r\n   at GraphQLParser.ParserContext.Dispose()\r\n   at GraphQLParser.Parser.Parse(ISource source)\r\n   at GraphQL.Execution.GraphQLDocumentBuilder.Build(String body) in C:\\projects\\graphql-dotnet\\src\\GraphQL\\Execution\\GraphQLDocumentBuilder.cs:line 20\r\n   at GraphQL.DocumentExecuter.ExecuteAsync(ExecutionOptions options) in C:\\projects\\graphql-dotnet\\src\\GraphQL\\Execution\\DocumentExecuter.cs:line 86\r\n   --- End of inner exception stack trace ---",
      "extensions": {
        "code": "APPLICATION",
        "codes": [
          "APPLICATION"
        ]
      }
    }
  ],
  "extensions": {
    "tracing": {
      "version": 1,
      "startTime": "2020-07-27T09:02:21.0125943Z",
      "endTime": "2020-07-27T09:02:21.0125943Z",
      "duration": 354200,
      "parsing": {
        "startOffset": 5200,
        "duration": 303400
      },
      "validation": {
        "startOffset": 0,
        "duration": 0
      },
      "execution": {
        "resolvers": []
      }
    }
  }
}
```

### PR DESCRIPTION
**Pull request by Shane32**
Fixes #80 

* Fixes an issue where trailing comments throw an exception
* Fixes an issue where comments can be applied to an incorrect node
* Unattached comments are returned in a list attached to the document

Note: these two documents parse the same, with the comment applied to `obj2`

```gql
{
  obj1 {
    field1
    # comment
  }
  obj2 {
    field2
  }
}
```

and

```gql
{
  obj1 {
    field1
  }
  # comment
  obj2 {
    field2
  }
}
```

### CODE REVIEW

File: src/GraphQLParser.Tests/ParserTests.cs [hunk 1]
```diff
@@ -12,6 +12,50 @@ public class ParserTests
...
+            // test
+            field = def.SelectionSet.Selections.Last() as GraphQLFieldSelection;
+            field.SelectionSet.Selections.Count().ShouldBe(1);
+            field.Comment.Text.ShouldBe("comment2");
+            // alt
+            subField = field.SelectionSet.Selections.First() as GraphQLFieldSelection;
+            subField.Comment.ShouldBeNull();
+            // extra document comments
+            document.UnattachedComments.Count().ShouldBe(3);
+            document.UnattachedComments[0].Text.ShouldBe("comment1");
```
Comment (Shane32) [inline src/GraphQLParser.Tests/ParserTests.cs:54]: Previously, `comment1` would have been incorrectly applied to `alt`.

**Comment by Shane32** [comment]
I have no idea why VS automatically reformatted the document.  Please use GitHub's "hide whitespace changes" feature to make the diff readable.
image

File: src/GraphQLParser.ApiTests/ApiApprovalTests.Public_Api_Should_Not_Change_Inadvertently.approved.txt [hunk 2]
```diff
@@ -82,6 +82,7 @@ namespace GraphQLParser.AST
         public GraphQLDocument() { }
         public System.Collections.Generic.List<GraphQLParser.AST.ASTNode>? Definitions { get; set; }
         public override GraphQLParser.AST.ASTNodeKind Kind { get; }
+        public System.Collections.Generic.List<GraphQLParser.AST.GraphQLComment>? UnattachedComments { get; set; }
```
Comment (sungam3r) [inline src/GraphQLParser.ApiTests/ApiApprovalTests.Public_Api_Should_Not_Change_Inadvertently.approved.txt:85]: @Shane32 Thanks. I am now looking at this simple and obvious solution and do not understand why I decided to throw an exception before.

**Comment by sungam3r** [comment]
I posted #106 to prevent unexpected changes.

**Comment by sungam3r** [comment]
I don't understand what happened to the formatting. Apparently, now it is correct. Anyway, looks good.

**Review by sungam3r — APPROVED** [review]

**Pull request merged**